Source/ProgressionSystemRuntime/Public/Data/PSSaveGameData.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#pragma once
+
+#include "Data/PSTypes.h"
+#include "Templates/Tuple.h"
+#include "GameFramework/SaveGame.h"
+#include "PSSaveGameData.generated.h"
+
+
+/**
+ * Defines the standard process for the saving slots names and index 
+ */
+UCLASS()
+class PROGRESSIONSYSTEMRUNTIME_API UPSSaveGameData : public USaveGame
+{
+	GENERATED_BODY()
+
+public:
+	/** Returns the name of the save slot. */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	static const FString& GetSaveSlotName();
+
+	/** Returns the Slot Index of the save slot. */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	static int32 GetSaveSlotIndex() { return 0; }
+
+	/** Returns the ProgressionRow by Index */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	const FPSRowData& GetSavedProgressionRowByIndex(int32 Index) const;
+
+	/** Set element by index */
+	UFUNCTION(BlueprintCallable, Category= "C++")
+	void SetCurrentProgressionRowByIndex(int32 InIndex);
+
+	/** Update the ProgressionRows map */
+	UFUNCTION(BlueprintCallable, Category = "C++")
+	void SetProgressionMap(const TMap<FName, FPSRowData> ProgressionRows);
+
+	/** Get current progression row */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	const FPSRowData& GetCurrentRow() const;
+
+	/** Get current progression row */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	FORCEINLINE FName GetCurrentRowName() const { return CurrentRowName; }
+
+	/** Set the current level by player tag*/
+	void SetRowByTag(FPlayerTag PlayerTag);
```

Comment:
Please try to make each your function exposed for blueprint developers because of open-source development: add UFUNCTION to this and other functions

Reply:
added

Reply:
You've added comment after UFUNCTION, but not BEFORE. It will definitely fail the cook.
Now:
```
UFUNCTION(BlueprintCallable, Category = "C++")
/** Set the current level by player tag*/
void SetRowByTag(FPlayerTag PlayerTag);
```
Has to be:
```
/** Set the current level by player tag*/
UFUNCTION(BlueprintCallable, Category = "C++")
void SetRowByTag(FPlayerTag PlayerTag);
```

Reply:
done